crates/unleash-edge-cli/src/lib.rs

```diff
@@ -483,6 +484,18 @@ pub struct CliArgs {
 
     #[clap(flatten)]
     pub otel_config: OpenTelemetryConfig,
+
+    // Internal usage only. Do not set this
+    #[clap(env = "EDGE_HOSTING", value_parser = hosting_type, hide = true)]
+    pub hosting_type: Option<Hosting>,
+}
```

Comment:
mfw this is the original implementation in the PR that introduced it, but I was [told to change it](https://github.com/Unleash/unleash-edge/pull/1086#discussion_r2262527786) 💀 

But yea, this is my preference too, and I like the hosting type function you've got under here 🙌🏼